app/src/app/components/sidebar/RecentMapsModal.tsx

```diff
@@ -32,11 +32,12 @@ export const RecentMapsModal = () => {
   const setMapDocument = useMapStore(store => store.setMapDocument);
   const [dialogOpen, setDialogOpen] = React.useState(false);
```

Comment:
H: Add an option in recent maps modal to filter for saved maps, to help cull down the long list?

```typescript
const [showAllMaps, setShowAllMaps] = React.useState(false);
const filteredMaps = showAllMaps
  ? userMaps
  : userMaps.filter(f => f.metadata?.find(f => f.key === 'saved'));
    
...
<Text as="label" size="2">
  <Flex as="span" gap="2">
    <Checkbox
      checked={showAllMaps}
      onClick={() => setShowAllMaps(prev => !prev)}
    />
    "Show all recent maps"
  </Flex>
</Text>
...
{filteredMaps.map((userMap, i) => (
              <RecentMapsRow
```